src/Laravel/Eloquent/State/LinksHandler.php

```diff
@@ -37,22 +40,72 @@ public function handleLinks(Builder $builder, array $uriVariables, array $contex
                 $identifier = $uriVariables[$uriVariable];
 
                 if ($to = $link->getToProperty()) {
-                    $builder = $builder->where($builder->getModel()->getTable().'.'.$builder->getModel()->{$to}()->getForeignKeyName(), $identifier);
+                    $builder = $builder->where($builder->getModel()->{$to}()->getQualifiedForeignKeyName(), $identifier);
 
                     continue;
                 }
 
                 if ($from = $link->getFromProperty()) {
                     $relation = $this->application->make($link->getFromClass());
-                    $builder = $builder->getModel()->where($builder->getModel()->getTable().'.'.$relation->{$from}()->getForeignKeyName(), $identifier);
+                    $builder = $builder->getModel()->where($relation->{$from}()->getQualifiedForeignKeyName(), $identifier);
 
                     continue;
                 }
 
-                $builder->where($builder->getModel()->getTable().'.'.$link->getIdentifiers()[0], $identifier);
+                $builder->where($builder->getModel()->qualifyColumn($link->getIdentifiers()[0]), $identifier);
             }
 
             return $builder;
+        } elseif ($operation instanceof GraphQlOperation) {
```

Comment:
you don't need the elseif as we `return $builder` above right?

Reply:
Probably since there only is HTTP and GraphQL Operations i guess. will fix it.